app/src/main/java/org/oppia/app/hintsandsolution/HintsAndSolutionDialogFragmentPresenter.kt

```diff
@@ -75,7 +75,7 @@ class HintsAndSolutionDialogFragmentPresenter @Inject constructor(
     }
 
     this.state = state
-    viewModel.newAvailableHintIndex.set(newAvailableHintIndex)
+    viewModel.newAvailableHintIndex.set((newAvailableHintIndex * 2))
```

Comment:
Should we add a comment above this line that why are we doing this ***2**?
This makes this more understandable

Reply:
Added:
```
    // The newAvailableHintIndex received here is coming from state player but in this implementation
    // hints/solutions are shown on every even index and on every odd index we show a divider.
    // Therefore multiplying the original index by 2.
```